src/components/molecules/ChatMessageBox/ChatMessageBox.tsx

```diff
@@ -85,8 +86,8 @@ export const ChatMessageBox: React.FC<ChatMessageBoxProps> = ({
   } = useShowHide();
 
   const addEmoji = useCallback(
-    (emoji: BaseEmoji) => {
-      if (emoji.native) {
+    (emoji: EmojiData) => {
+      if ("native" in emoji && emoji.native) {
```

Comment:
Because TypeScript says no.

emoji is a union of 2 types, and only one of them has native defined on it. So this is a way of narrowing the types and telling typescript to only run this on the type that has it. The second part of the check is probably irrelevant I think, but it’s the logic that existed previously so didn’t want to deal with figuring out any potential nuances of changing it.

Reply:
Note that i'm pretty certain I tried `if (emoji?.native)` before figuring out this method, and it doesn't work because the 2 types within `EmojiData` don't both have the `native` field as part of a common 'base type' (which is what we do in a few cases, and is how we have tended to work around/avoid a situation like this in our own code, rightly or wrongly)